### PR TITLE
Don't set active state to newly created HPA

### PR DESCRIPTION
**Pull request by orangedeng**
**Problem:**
The hpa object needs conditions to indicate the state,
if there are not any, we should set it to initializing.

Related issue:
https://github.com/rancher/rancher/issues/20955

Related PR:
https://github.com/rancher/types/pull/907

### CODE REVIEW

**Comment by gitlawr** [comment]
Can you add some tests accordingly

**Comment by gitlawr** [comment]
LGTM

**Comment by gitlawr** [comment]
Note: this pr is rebased on https://github.com/rancher/rancher/pull/20937 to address conflicts, so needs rebasing after the dependency is merged.

**Comment by orangedeng** [comment]
Updated

File: vendor.conf [hunk 1]
```diff
@@ -74,3 +74,4 @@ google.golang.org/api/admin/directory/v1      d827405e6070b8f323fa8681174e93f019
 golang.org/x/oauth2                           aaccbc9213b0974828f81aaac109d194880e3014
 github.com/golang/oauth2/google               9f3314589c9a9136388751d9adae6b0ed400978a
 go.opencensus.io                              v0.21.0                                  transitive=true
+k8s.io/kube-aggregator                        kubernetes-1.12.2                        https://github.com/kubernetes/kube-aggregator
```
Comment (alena1108) [inline vendor.conf]: nit: new line

**Comment by alena1108** [comment]
LGTM

@orangedeng can you split it to 2 commits - code and vendor changes.

**Comment by orangedeng** [comment]
Yes. This PR has dependency on `https://github.com/rancher/rancher/pull/20937`. Will update this PR when 20937 PR is merged.

**Comment by orangedeng** [comment]
Rebased on master.

**Pull request merged**